river/command/cursor.zig

```diff
@@ -0,0 +1,58 @@
+// This file is part of river, a dynamic tiling wayland compositor.
+//
+// Copyright 2022 The River Developers
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+const std = @import("std");
+
+const util = @import("../util.zig");
+
+const server = &@import("../main.zig").server;
+
+const Config = @import("../Config.zig");
+const Error = @import("../command.zig").Error;
+const Seat = @import("../Seat.zig");
+
+pub fn cursor(
+    _: *Seat,
+    args: []const [:0]const u8,
+    _: *?[]const u8,
+) Error!void {
+    if (std.mem.eql(u8, "jitter", args[1])) {
+        if (args.len < 3) return Error.NotEnoughArguments;
+        switch (args.len) {
+            3 => {
+                server.config.cursor_hide_jitter = try std.fmt.parseInt(u32, args[2], 10);
+                server.config.cursor_unhide_jitter = server.config.cursor_hide_jitter;
+            },
+            4 => {
+                server.config.cursor_hide_jitter = try std.fmt.parseInt(u32, args[2], 10);
+                server.config.cursor_unhide_jitter = try std.fmt.parseInt(u32, args[3], 10);
+            },
+            else => return Error.TooManyArguments,
+        }
+    } else if (std.mem.eql(u8, "timeout", args[1])) {
+        if (args.len < 3) return Error.NotEnoughArguments;
+        if (args.len > 3) return Error.TooManyArguments;
+        server.config.cursor_hide_timeout = try std.fmt.parseInt(u31, args[2], 10);
```

Comment:
If the timeout is currently 0 and `riverctl hide-cursor timeout 1000` is run but the cursor is not moved, the cursor is never hidden since we don't actually start the timer here.

Likewise, if the cursor is hidden with an absurdly high `unhide_jitter` it is currently impossible to unhide the cursor through setting a lower `unhide_jitter` and moving the cursor. The only way to get the cursor shown again current is to click.

I think both of these should be resolved by calling `Cursor.unhide()` for the cursor of every seat in `server.input_manager.seats` at the end of this function.

Reply:
I solved this by iterating over seats and calling `cursor.unhide()` to reset the timeout, this seems like the simplest solution.